src/views/Home.jsx

```diff
@@ -1,12 +1,68 @@
 import './Home.css';
 import { SingleList } from '../components/SingleList.jsx';
+import { useState } from 'react';
+import { createList } from '../api/firebase.js';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ data, userId, userEmail, setListPath }) {
+	const [newListName, setNewListName] = useState('');
+
+	const navigate = useNavigate();
+
+	const handleChange = (e) => {
+		setNewListName(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const currentLists = data.map((list) => {
+			return list.name;
+		});
+
+		if (newListName.includes('/')) {
+			alert(
+				'Invalid list name. "/" is an unsupported character, please try a different list name.',
+			);
+			return;
+		}
+		//Check if the new list has a title that already exists in db
+		else if (currentLists.includes(newListName)) {
+			alert('List already exists. Please enter a unique list name.');
+		} else {
+			//Create new list
+			const response = await createList(userId, userEmail, newListName);
+
+			//If list creation successful, show alert and navigate to new list view
+			if (response) {
+				alert('List created successfully.');
+				setListPath(`${userId}/${newListName}`);
+				navigate('/list');
+				//If list creation error, show alert
+			} else {
+				alert('List could not be created.');
+			}
+		}
+
+		setNewListName('');
```

Comment:
nit: This is not necessary to update. I just want to share based on my experience. My manager told me that it is better to avoid nested `if-else` because it might be more difficult to read. We can refactor to my example below.
The idea of this refactor:

1. Two First `if-else` statements are just `guard clauses`, which I usually call like that. Therefore, is not a main function. I will separate them into their own if statements. If it's true, stop the function by calling `return`
2. If the parameter passes the `guard clause`, so we will run the main function, which is in the else clause before.

```jsx
	const handleSubmit = async (e) => {
		e.preventDefault();

		if (newListName.includes('/')) {
			alert(
				'Invalid list name. "/" is an unsupported character, please try a different list name.',
			);
			return;
		}

		const currentLists = data.map((list) => {
			return list.name;
		});
		if (currentLists.includes(newListName)) {
			alert('List already exists. Please enter a unique list name.');
			return;
		}

		const response = await createList(userId, userEmail, newListName);
		if (response) {
			alert('List created successfully.');
			setListPath(`${userId}/${newListName}`);
			navigate('/list');
		} else {
			alert('List could not be created.');
		}

		setNewListName('');
	};
```

Reply:
Oh that's a great way to refactor it. Thanks for the info Ray. I hadn't known the vocabulary for `guard clause` and that made it easier to conceptualize. I've gone ahead and pushed the refactored code.